### PR TITLE
Reset L4 DualStack Gauge metrics before exporting

### PR DESCRIPTION
**Pull request by panslava**
Now, if service does not exist anymore, it will not be zeroed in metrics

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -626,12 +626,30 @@ func (im *ControllerMetrics) computeL4ILBMetrics() map[feature]int {
...
+					IPFamilies:     ipFamilies,
+					IPFamilyPolicy: ipFamilyPolicy,
+					Status:         status,
+				}] = 0
+			}
+		}
+	}
+	return zeroCounts
+}
+
```
Comment (cezarygerard) [inline pkg/metrics/metrics.go:646]: move it to the bottom of the file, or to another file "l4_dualstack_metrics.gom" (together with all dualstack stuff)

File: pkg/metrics/l4metrics_test.go [hunk 2]
```diff
@@ -467,8 +467,12 @@ func TestComputeL4NetLBDualStackMetrics(t *testing.T) {
 			for i, serviceState := range tc.serviceStates {
 				newMetrics.SetL4NetLBDualStackService(fmt.Sprint(i), serviceState)
 			}
+			expectedCount := zeroL4DualStackCounts()
```
Comment (cezarygerard) [inline pkg/metrics/l4metrics_test.go:470]: maybe add short comment how you construct the expected count

Comment (cezarygerard) [inline pkg/metrics/l4metrics_test.go:470]: this funny  code will work either:

```
		t.Run(tc.desc, func(t *testing.T) {
			t.Parallel()
			newMetrics := FakeControllerMetrics()
			for i, serviceState := range tc.serviceStates {
				newMetrics.SetL4ILBDualStackService(fmt.Sprint(i), serviceState)
			}
			for key, _ := range zeroL4DualStackCounts() {
				tc.expectL4ILBDualStackCount[key] = tc.expectL4ILBDualStackCount[key]
			}
			got := newMetrics.computeL4ILBDualStackMetrics()
			if diff := cmp.Diff(tc.expectL4ILBDualStackCount, got); diff != "" {
				t.Fatalf("Got diff for L4 ILB Dual-Stack service counts (-want +got):\n%s", diff)
			}
		})
```

which is better?

Comment (panslava) [reply pkg/metrics/l4metrics_test.go:470]: yeh, good point, it saves one line, but I prefer not to change tc, though I understand, this code looks a bit strange maybe
i will add comment

Comment (panslava) [reply pkg/metrics/l4metrics_test.go]: done

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (panslava) [reply pkg/metrics/metrics.go]: moved to another file

**Comment by panslava** [comment]
/hold

I will squash commits after lgtm

**Comment by cezarygerard** [comment]
just use  Reset()  method on gave metrics :-D

**Comment by panslava** [comment]
@cezarygerard I changed, now using Reset

**Pull request merged**